frontend/src/app/shared/directives/focus/focus-helper.ts

```diff
@@ -34,7 +34,12 @@ export class FocusHelperService {
   public focus(element:HTMLElement):void {
     setTimeout(() => {
       const focusable = findAllFocusableElementsWithin(element)[0] || element;
-      focusable?.focus();
+      if (window.location.hash) {
+        focusable?.focus();
+      } else {
+        // Prevent scroll if there's no hash in the URL
+        focusable?.focus({ preventScroll: true });
+      }
```

Comment:
I did not test this, but this whole block could probably be changed to `focusable?.focus({ preventScroll: !!window.location.hash });`